src/server/handlers.rs

```diff
@@ -521,6 +521,102 @@ impl<P: Publisher> Handler<P> {
+        let mut existing_provider_refs: HashMap<String, String> = HashMap::new();
+        for model_summary in stored_models.iter() {
+            if let Some(deployed_version) = &model_summary.deployed_version {
+                let (stored_manifest, _) = match self
+                    .store
+                    .get(account_id, lattice_id, &model_summary.name)
+                    .await
+                {
+                    Ok(Some(m)) => m,
+                    Ok(None) => (StoredManifest::default(), 0),
+                    Err(e) => {
+                        error!(error = %e, "Unable to fetch data");
+                        self.send_error(msg.reply, "Internal storage error".to_string())
+                            .await;
+                        return;
+                    }
+                };
+                if stored_manifest.name() != name {
+                    if let Some(deployed_manifest) = stored_manifest.get_version(deployed_version) {
+                        for component in deployed_manifest.spec.components.iter() {
+                            if let Properties::Capability {
+                                properties:
+                                    CapabilityProperties {
+                                        image: image_name, ..
+                                    },
+                            } = &component.properties
+                            {
+                                let image_ref_split: Vec<&str> = image_name.split(':').collect();
+                                if let (Some(&ref_link), Some(&ref_version)) =
+                                    (image_ref_split.first(), image_ref_split.last())
+                                {
+                                    existing_provider_refs
+                                        .insert(ref_link.to_string(), ref_version.to_string());
+                                }
+                            }
+                        }
+                    };
+                }
+            }
+        }
+
+        let staged_model = match req.version.clone() {
+            Some(v) => {
+                if let Some(model) = manifests.get_version(&v) {
+                    model
+                } else {
+                    manifests.get_current()
+                }
+            }
+            None => manifests.get_current(),
+        };
+
+        // Compare if any of the provider refs in the staged model are duplicates
+        for component in staged_model.spec.components.iter() {
+            if let Properties::Capability {
+                properties:
+                    CapabilityProperties {
+                        image: image_name, ..
+                    },
+            } = &component.properties
+            {
+                let image_ref_split: Vec<&str> = image_name.split(':').collect();
+                if let (Some(&ref_link), Some(&ref_version)) =
+                    (image_ref_split.first(), image_ref_split.last())
```

Comment:
This shows up in a place or two, if it makes sense consider a function like
```
fn parse_image_ref_and_tag(image_name: &str) -> Option<(String, String)>
```
I'm bad at naming but you get the idea 😄